sklearn/linear_model/_sgd_fast.pyx

```diff
@@ -360,16 +360,16 @@ cdef class SquaredEpsilonInsensitive(Regression):
         return SquaredEpsilonInsensitive, (self.epsilon,)
 
 
-def _plain_sgd(cnp.ndarray[double, ndim=1, mode='c'] weights,
+def _plain_sgd(double[::1] weights,
```

Comment:
If `weights` is not modified, it must be `const`-qualified.
```suggestion
def _plain_sgd(const double[::1] weights,
```

This comments also applies to all other memoryview which aren't modified _except_ memory-view using fused types due to limitation present in Cython<3; hence this should apply to `average_weights` and `validation_mask`.

See discussions in https://github.com/scikit-learn/scikit-learn/issues/10624.